src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs

```diff
@@ -101,6 +101,19 @@ public async Task Invoke(HttpContext context)
         }
         catch (Exception ex)
         {
+
+            if (ex is OperationCanceledException operationCanceledException && context.RequestAborted.IsCancellationRequested)
+            {
+                _logger.LogDebug("Request aborted");
+
+                if (context.Response.HasStarted)
```

Comment:
Did you meant to write this? Can we change the status code if the response has started?
```suggestion
                if (!context.Response.HasStarted)
```

Reply:
Please see @Tratcher  https://github.com/dotnet/aspnetcore/issues/44697#issuecomment-1367482369

Reply:
Kahbazi is right, see the code https://source.dot.net/#Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpProtocol.cs,240

Reply:
Sorry, my original comment wasn't clear about polarity.